### DISCUSSION TITLE
15 giugno 2022 - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#12] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/e818fed4-29b9-4716-9f08-92bb91f1f807)

**Comment by IonutCicio** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/2da85294-3aa1-42d6-8cf6-adea3e51d382)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/79eaa1ee-d964-499b-8eb6-c1b4bc06199f)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/4b0e3441-abe3-4ae6-9ad0-4fdbfab0ecf5)

**Reply by IonutCicio** [reply]
L'algoritmo che ho scritto conferma i risultati

```python
exams/2022-06-15.pdf
Keys: {{'A', 'C'}, {'A', 'H'}, {'B', 'H'}, {'E', 'H'}}
Is 3NF? false
A decomposition is: {{'A', 'B', 'C'}, {'A', 'C', 'H'}, {'A', 'E', 'H'}, {'B', 'C', 'E'}, {'B', 'D'}, {'C', 'H'}, {'D', 'E'}}
```

**Reply by Alessandro700** [reply]
Buongiorno, mi si è creato un dubbio, sul passaggio di una copertura minimale, e precisamente nel passo 2 dell'algoritmo, dove proviamo a ridurre le parti sinistre, ABC-->H a me risulta che non si può ridurre... quindi poi cambia un po tutto il risultato. provo a condividera la mia soluzione, ovviamente se lo è un piccolo errore che ora mi ha creato un dubbio. Grazie 
<img width="1608" height="773" alt="image" src="https://github.com/user-attachments/assets/0fab46ca-a3a8-4c2d-9cd5-b997846ce9f8" />
<img width="924" height="592" alt="image" src="https://github.com/user-attachments/assets/4289fcba-f05f-4482-a482-a1bd03cd7d4a" />

**Reply by Alessandro700** [reply]
<img width="1579" height="671" alt="image" src="https://github.com/user-attachments/assets/40ae521a-fb73-455e-bbf3-e14d9d488bde" />

**Reply by flqmex** [reply]
> Buongiorno, mi si è creato un dubbio, sul passaggio di una copertura minimale, e precisamente nel passo 2 dell'algoritmo, dove proviamo a ridurre le parti sinistre, ABC-->H a me risulta che non si può ridurre... quindi poi cambia un po tutto il risultato. provo a condividera la mia soluzione, ovviamente se lo è un piccolo errore che ora mi ha creato un dubbio. Grazie 


Ciao, per rendere la tua copertura minimale corretta, basta notare che la dipendenza $$ABC \to H$$ è riducibile in $$AC \to H$$.
Questa riduzione è possibile perché:
 - l'insieme $$AC$$ è un sottoinsieme proprio di $$ABC$$: $$AC \subset ABC$$
 - $$H$$ fa parte della chiusura di $$AC$$ in $$F_{min}$$
 
Ora se procedi con la tua soluzione, otterrai una decomposizione esatta.

**Reply by Alessandro700** [reply]
Quindi va controllato pure queste cose? Ma è un po raro o sbaglio? Ne sto facendo di esami e fino adesso mai come questo… con questa particolarità 
ps sto facendo un esame dove ho notato questa particolarità, ora mi è piu chiara, grazie per la spiegazione
